project_task_code/models/project_task.py

```diff
@@ -27,3 +27,15 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.multi
+    @api.depends('name', 'code')
+    def name_get(self):
+        result = super(ProjectTask, self).name_get()
+        new_result = []
+
+        for task in result:
+            rec = self.browse(task[0])
+            name = '%s - %s' % (rec.code, rec.name)
```

Comment:
Can you change this to:

`name = '%s - %s' % (rec.code, task[1])`

So other modules can still append other strings to it.